### ISSUE TITLE
Possible to add support for alt filenames?

### ISSUE THREAD

**Issue by Code-Slave** [ankenyr/jellyfin-youtube-metadata-plugin#40]
I use a youtube archiver called tubearchivist. Really nice app that id linke to get into jellyfin. However it uses the following to save files

20170509_cdai_HkKpyM_DeWalt DWE6423 Random Orbit Sander vs Hyper Tough.mp4

Is it feasable to make the placement and or delimeters for the id configurable or add a switch for tubearchivist?

**Comment by Code-Slave** [comment]
yes

**Comment by HStep20** [comment]
I'd like to echo this request as well. 

Tubesync (sonarr for youtube) lets you define naming conventions for playlists/channels as you download them, which would be nice to match this layout with. However, it becomes an issue when someone decides to migrate over with terabytes of already-downloaded media. On top of that, tubesyncs current state is pretty unreliable and has not updated in a while. 

I know for some, myself included, TubeArchivist has taken over in it's use case but It only supports its one naming convention. I think supporting alternate filename patterns with the plugin would be helpful when it comes to importing existing archive libraries into Jellyfin.

**Comment by Code-Slave** [comment]
Exactly. I was pushing hard for tubesync but it just stalled.

**Comment by kzshantonu** [comment]
I vote for this too. I'm using tubearchivist and would love to add the videos to jellyfin

**Comment by ankenyr** [comment]
Sorry I let this one linger.
Is the desire here to have individual regexes for popular downloader helpers or providing your own regex? The later is stronger but does open things up to being DOS'ed by anyone with admin by putting in an expensive regex.
https://en.wikipedia.org/wiki/ReDoS
The former would be exhaustive to continue to add to since yt-dlp itself allows you to make any file name you desire with templates.

**Comment by kzshantonu** [comment]
That's true. In tubearchivist's case though, the files [have to be named](https://github.com/tubearchivist/tubearchivist/blob/bb0f3dad1a02a375a306cc65b1689de92b6fc8e8/README.md?plain=1#L200) the way it saves them. I think the former option is good for now, i.e. adding support based on user votes and/or popularity

**Comment by ankenyr** [comment]
Does TubeArchivist also download the channels info.json? As an example I have a file named
`Ahoy - NA - Ahoy_-_Videos [UCE1jXbVAGJQEORz9nZqb5bQ].info.json`
and the contents are things like the channels thumbnails, its epoch, description and other information.

**Comment by HStep20** [comment]
Unfortunately, it looks like it does not download any extra files alongside the videos, which is a shame and something I hadn't realized beforehand. It embeds metadata into the file, but doesnt have any option to write the json as a sidecar in the same folder. 

Another big thing I overlooked/didnt realize as I had hopped back to tubesync, was that it doesnt divide anything up into folders. It has an incredibly static file system that will apparently never change where it dumps all videos of a channel into one directory. This means that individual playlists are jumbled up together with every other video from a content creator when they are downloaded regardless of if you specify one/multiple playlists to download. Im not sure this will implement nicely into the jellyfin interface, as it will see each 'creator' folder as a one season monolith, and will be unable to break them up into 'seasons' based on playlists. 

Of course, its great for people who archive entire channels, which I do have a few of, but at the same time it has 0 support for getting specific playlists from content creators, since it jumbles them all up into one folder and doesnt write any sidecar data. Perhaps releasing the update to fix #47 while reapproaching this would be a good idea.

**Comment by ankenyr** [comment]
Yea I am going to revert the changes I made tonight and release for #47. I want to take a look at Tube Archivist for my own personal use. Already opened a issue on their github which will block me from moving. What made you move to Tubesync from Tube Archivist? Tube Archivist having an API with all the info I would need should make things simpler to  work with. I have not looked at the current state of TubeSync to know if they have an API created or on their roadmap.

**Comment by HStep20** [comment]
Ive gone back and forth between them, because I think they both have their uses, but the biggest problem with TubeSync is that it just straight up doesnt work right. Even though I much prefer it because of the flexibility with naming, Ive had on and off issues where it wouldnt download videos it had set up to sync and it just stalls out with scheduled tasks. Right now for example, I have 111 scheduled tasks that are supposed to be running, but they never run. My last video download was a month ago. My other big issue is that it doesnt support SponsorBlock, which TubeArchivist does. I hate ads of any kind, so seeing sponsored segments in my downloaded videos really puts me off. The update cycle is slow, and while it looks like the dev is kind of active, new features are few and far between. The latest update consisting of 'bug fixes and minor features' was 3 days ago - the one before it came out 10 months ago. 

Lots of 'putting out fires' in the issues, but no real progress to resolve the database locks, lack of sponsor block integration, extra naming convention tags supported by yt-dlp, and other features that are desperately needed.

I keep going back and forth because TubeArchivist is better, but TubeSync actually does what I want/need for my setup.

**Comment by Code-Slave** [comment]
I basically mirror Hstep20s thoughts. TA works well but very opinionated on things. I actually left the community because I am an archiver and kept requesting things that made that easier and he stated My thoughts didnt match his plans and asked meto stop. Shrug.  Tubesync naming flexibility with TA capabilities would be IDEAL. I did the switcharoo for a long time but TS just wouldnt work and dev is kinda stalled.

**Comment by ankenyr** [comment]
Mind expanding? Got links to issues you filed?I would be interested in
hearing more. I am on the jellyfin matrix server if you would rather chat
over that.

**Comment by bandundu** [comment]
Hi, I would also greatly benefit from such a feature. I had a script that would create renamed symlinks with the correct naming scheme in a seperate top folder. It worked like a charm but I somehow deleted it. I might look into recreating it  and sharing it here.

**Comment by SpaceAgeHero** [comment]
I'm using [YoutubeDL-Material ](https://github.com/Tzahi12345/YoutubeDL-Material).

It has downloaded all my playlists and single videos without caring for video IDs in the filenames.

File scheme is like this:

> Big Buck Bunny 60fps 4K - Official Blender Foundation Short Film.mp4
Big Buck Bunny 60fps 4K - Official Blender Foundation Short Film.webp
Big Buck Bunny 60fps 4K - Official Blender Foundation Short Film.info.json

I've set up the plugin in Jellyfin but it doesn't seem to recognize anything. I want to use only local metadata and webp images.
How can this be done?

**Comment by Code-Slave** [comment]
@SpaceAgeHero   look at ytdl-sub. Ive switch to that for everything. builds nfo files so no worries for metadata etc in jellyfin

**Comment by SpaceAgeHero** [comment]
@Code-Slave Thanks for the heads up but ytdl-sub lacks a neat web interface.

**Comment by BobbyWibowo** [comment]
Has anybody got any recent updates regarding this specific use case?
